testdata/p4_16_samples/psa-portid-using-newtype2.p4

```diff
@@ -0,0 +1,293 @@
+#include "core.p4"
+
+// From psa.p4
+
+#define PORT_BITWIDTH 9  // target-specific
+
+// PortIdUInt_t is only intended for casting values from type PortId_t
+// to an unsigned integer with the same width.
+typedef bit<PORT_BITWIDTH> PortIdUInt_t;
+
+// request translation by P4Runtime agent of all occurrences of PortId_t exposed
+// to control-plane
+@p4runtime_translate("port")
+newtype bit<PORT_BITWIDTH> PortId_t;
+
+//const PortId_t PSA_CPU_PORT = (PortId_t) 9w192; // target-specific
+// With (PortId_t) cast on unsigned int literal, I get this error from
+// newtype branch of p4c:
+//psa-portid-using-newtype1.p4(16): error: cast: Cannot evaluate initializer for constant
+
+//const PortId_t PSA_CPU_PORT = 9w192;  // target-specific
+// With no cast on unsigned int literal, I get this error:
+//psa-portid-using-newtype1.p4(14): error: PSA_CPU_PORT: Cannot unify bit<9> to PortId_t
+//newtype bit<9> PortId_t;
+//                        ^
+//psa-portid-using-newtype1.p4(14)
+//newtype bit<9> PortId_t;
+//               ^^^^^^^^
+
+// This also gives 'Cannot evaluate initializer for constant'.
+//const PortId_t PSA_CPU_PORT = ((PortId_t) ((bit<9>) 9w192));  // target-specific
+
+// This is intended as a workaround until 'const' can be used for
+// PSA_CPU_PORT, or I suppose it could be a long term solution, too.
+#define PSA_CPU_PORT ((PortId_t) 9w192)
+
+// The width of this is _not_ target-specific.  All targets will use
+// this bit width.  This type is only intended for copying data-plane
+// number space PortId_t values into headers to be sent to the CPU
+// port, or for parsing headers containing PortId_t values in certain
+// headers in packets from the CPU port.
+@p4runtime_translate("port")
+newtype bit<32> PortIdInHeader_t;
```

Comment:
Same comment as on the annotation to PortId_t. I do not believe it should be necessary at all.